signature/policy_config.go

```diff
@@ -51,28 +52,44 @@ func (err InvalidPolicyFormatError) Error() string {
 // NOTE: When this function returns an error, report it to the user and abort.
 // DO NOT hard-code fallback policies in your application.
 func DefaultPolicy(sys *types.SystemContext) (*Policy, error) {
-	return NewPolicyFromFile(defaultPolicyPath(sys))
+	policyPath, err := defaultPolicyPath(sys)
+	if err != nil {
+		return nil, err
+	}
+	return NewPolicyFromFile(policyPath)
 }
 
 // defaultPolicyPath returns a path to the default policy of the system.
-func defaultPolicyPath(sys *types.SystemContext) string {
-	return defaultPolicyPathWithHomeDir(sys, homedir.Get())
+func defaultPolicyPath(sys *types.SystemContext) (string, error) {
+	policyFilePath, err := defaultPolicyPathWithHomeDir(sys, homedir.Get())
+	if err != nil {
+		return "", err
+	}
+	return policyFilePath, nil
 }
 
 // defaultPolicyPathWithHomeDir is an internal implementation detail of defaultPolicyPath,
 // it exists only to allow testing it with an artificial home directory.
-func defaultPolicyPathWithHomeDir(sys *types.SystemContext, homeDir string) string {
+func defaultPolicyPathWithHomeDir(sys *types.SystemContext, homeDir string) (string, error) {
+	var checkedPolicyPaths []string // To store the paths being checked
+
 	if sys != nil && sys.SignaturePolicyPath != "" {
-		return sys.SignaturePolicyPath
+		return sys.SignaturePolicyPath, nil
 	}
+
 	userPolicyFilePath := filepath.Join(homeDir, userPolicyFile)
+	checkedPolicyPaths = append(checkedPolicyPaths, userPolicyFilePath)
```

Comment:
A microoptimization: These things can be recorded only after the file is determined to be missing.

… or, alternatively, don’t compute a `[]string` on the fly at all: We know that if we get to the end, we have tried exactly two files, and we already have variables with the full paths; so we can construct the error message directly without `strings.Join`.

Reply:
It makes sense.. Thanks for your review comments. I removed `strings.Join` instead used existing variable names.. as per availability.